src/middlewared/middlewared/api/base/server/ws_handler/rpc.py

```diff
@@ -216,15 +217,29 @@ async def process(self, origin: Origin, ws: WebSocketResponse):
                     )
                     break
 
-                if not app.authenticated and len(msg.data) > 8192:
+                try:
+                    message = msg_size_check(app.authenticated, len(msg.data), msg.data)
+                except MsgSizeError as err:
+                    if err.limit is MsgSizeLimit.UNAUTHENTICATED:
+                        await ws.close(
+                            code=WSCloseCode.INVALID_TEXT,
+                            message='Anonymous connection max message length is 8 kB'.encode('utf-8'),
+                        )
+                        break
+
+                    creds = app.authenticated_credentials.dump() if connection.authenticated_credentials else None
+                    origin = app.origin.repr() if connection.origin else None
+
+                    self.middleware.logger.error(
+                        'Client using credentials [%s] at [%s] sent message with payload size [%d bytes] '
+                        'exceeding limit of %d for method %s',
+                        creds, origin, err.datalen, err.limit, err.method_name
+                    )
                     await ws.close(
-                        code=WSCloseCode.INVALID_TEXT,
-                        message=b"Anonymous connection max message length is 8 kB",
+                        code=WSCloseCode.MESSAGE_TOO_BIG,
+                        message='Max message length is 64 kB'.encode('utf-8'),
```

Comment:
We can raise `MsgSizeError` with proper `message`. ANother advantage of this is that we won't be hard-coding duplicate message size constants here.